log4j-api/src/main/java/org/apache/logging/log4j/util/LoaderUtil.java

```diff
@@ -106,30 +104,31 @@ public ClassLoader run() {
     }
 
     public static ClassLoader[] getClassLoaders() {
-        final List<ClassLoader> classLoaders = new ArrayList<>();
+        final Collection<ClassLoader> classLoaders = new LinkedHashSet<>();
```

Comment:
LinkedHashSet is more expensive in environments with few values, but performs better in the worst case without sacrificing order.